1.6/contrib/configuration/jobs/OpenShift Sample/config.xml

```diff
@@ -0,0 +1,153 @@
+<project>
+  <actions/>
+  <description></description>
+  <keepDependencies>false</keepDependencies>
+  <properties>
+    <hudson.model.ParametersDefinitionProperty>
+      <parameterDefinitions>
+        <hudson.model.TextParameterDefinition>
+          <name>CREDENTIALS</name>
+          <description>Output of 'oc config view --minify --flatten' when logged in as a user with permission to run builds and deployments in your OpenShift project.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.TextParameterDefinition>
+        
+        <hudson.model.TextParameterDefinition>
+          <name>PROJECT</name>
+          <description>The OpenShift project this job will access</description>
+          <defaultValue>test</defaultValue>
+        </hudson.model.TextParameterDefinition>
+        
+        <hudson.model.TextParameterDefinition>
+          <name>BUILD_CONFIG</name>
+          <description>The BuildConfig to trigger</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+        <hudson.model.TextParameterDefinition>
+          <name>SERVICE</name>
+          <description>The service to test after build completion</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+        <hudson.model.TextParameterDefinition>
+          <name>DEPLOYMENT_CONFIG</name>
+          <description>The name of the DeploymentConfig that deploys the output of the BuildConfig</description>
+          <defaultValue>frontend</defaultValue>
+        </hudson.model.TextParameterDefinition>
+
+      </parameterDefinitions>
+    </hudson.model.ParametersDefinitionProperty>
+  </properties>
+  <scm class="hudson.scm.NullSCM"/>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers/>
+  <concurrentBuild>false</concurrentBuild>
+  <builders>
+    <hudson.tasks.Shell>
+      <command>
+if [ -z "$CREDENTIALS" ]; then
+  SERVER=https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}
+  TOKEN=`cat /var/run/secrets/kubernetes.io/serviceaccount/token`
+  oc config set-cluster local-cluster --server=$SERVER --insecure-skip-tls-verify=true
+  oc config set-credentials local-user --token=$TOKEN
+  oc config set-context local --cluster=local-cluster --user=local-user --namespace=$PROJECT
+  oc config use-context local
+else 
+  export KUBECONFIG=.kubeconfig
+  echo "${CREDENTIALS}" > "${KUBECONFIG}"
+fi
+
+oc project ${PROJECT}
+
+# Switch back to .clusterIP once there is a new openshift release where we can get the updated oc binary 
+#TEST_ENDPOINT=`oc get service ${SERVICE} -t '{{.spec.clusterIP}}{{":"}}{{ $a:= index .spec.ports 0 }}{{$a.port}}'`
+TEST_ENDPOINT=`oc get service ${SERVICE} -t '{{.spec.portalIP}}{{":"}}{{ $a:= index .spec.ports 0 }}{{$a.port}}'`
```

Comment:
only if you're going to guarantee me the app in question setup a headless service, right?

in theory i could use the normal service DNS, but even that assumes the user got DNS working correctly, which i might rather not assume (especially if you consider they could be running this jenkins server outside of openshift, or from another openshift deployment).  so this method seems more robust.